### ISSUE TITLE
ToC2: cell sizes are widened 

### ISSUE THREAD

**Issue by RinaldoB** [ipython-contrib/jupyter_contrib_nbextensions#871]
Hi, turning on the Table of Contents extension has an side-effect for me. The width of the cells is extended to fit the browser window. Without ToC2 the width of the cells is limited beyond a certain size.

**Comment by jfbercher** [comment]
This is not really an issue: this behavior is intended, so to have the sidebar to display nicely. 

We could add another configuration option for that if people find it sufficiently useful but this will further complicate the code (have to test if the toc2 is displayed or not, floating or displayed as a sidebar).

**Comment by RinaldoB** [comment]
I support the option to widen the cells only if the ToC is visible.

The current behavior alters the notebook display, only if the ToC2 extension is enabled. Personally I intend to use a ToC only a couple of my notebooks, however all of the notebooks are affected by this, so I'm forced to turn the extension off.

**Comment by odedbd** [comment]
I find the widening effect generally helpful. If it ends up being applied only when the ToC is visible it would be great if "wide_notebook" could be available as a standalone extension

**Comment by jcb91** [comment]
yep, making a standalone is certainly possible, and sounds reasonable

**Comment by jfbercher** [comment]
An attempt to address this is on #887. 

For a standalone extension, something like this
`$("#view_menu").append("<li id=\"toggle_width\" title=\"Enlarge notebook\"><a href=\"javascript:{var percent = $('#notebook-container').width()/$('#notebook-container').parent().width()*100; var new_width = percent < 90? '98%': '82%'; $('#notebook-container').css({'width':new_width, 'margin-left':'auto'})}\">(Un)stretch notebook</a></li>")` in `custom.js` should do the trick.

**Comment by jcb91** [comment]
or even just

```css
#notebook-container {
    width: auto;
}
```

in `custom.css`!

**Comment by KadeG** [comment]
Just spent a bit tracking down what was widening my cells. I wish it mentioned it in the readme. It makes for difficult reading in text-heavy notebooks.

**Comment by jfbercher** [comment]
@KadeG  See #887. In addition I just updated the README :-)

**Comment by jfbercher** [comment]
Closed by #887.

**Comment by KadeG** [comment]
@jfbercher You're awesome!

**Comment by fkjogu** [comment]
For me, the cells are still widened even if I have switched this off in the options.

Using jupyter 4.3.0, Python 3, and whatever is current in PIP.

**Comment by fkjogu** [comment]
Also, when disabling the sidebar, the cells shrink again, but are not centered anymore.

**Comment by fkjogu** [comment]
I've created a pull request #1091

**Comment by jfbercher** [comment]
Thanks for reporting and looking at that. However, I can't reproduce... ??
Can you try with the version in the repo (just download main.js and toc2.js)?

When toggling the corresponding option in nbconfigurator and reloading the notebook, the behavior is as expected (widen cell if checked and normal if not). of course when the sidebar is present, the cells are widen. 
For the centering problem, this was corrected in a recent commit, see [toc2#L167](https://github.com/ipython-contrib/jupyter_contrib_nbextensions/blob/master/src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.js#L167) and  [toc2#L175](https://github.com/ipython-contrib/jupyter_contrib_nbextensions/blob/master/src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.js#L175).

**Comment by fkjogu** [comment]
That's weird, those lines were definitly not there, when I was making the pull request. That certainly fixes the centering.

But is it expected behaviour, that the notebook is widened when a sidebar is present, no matter what was selected in the options?

What my pull request does is, it really only widens, if the option is checked. And it just makes room, when space is needed for the sidebar, even if not in wide mode.

**Comment by jfbercher** [comment]
> But is it expected behaviour, that the notebook is widened when a sidebar is present, no matter what was selected in the options?

Yes. This is the expected behaviour. 
I find that displaying the sidebar on the extreme left when there is a large space available is not very pretty... However, this can be discussed, and since I personally use widenNotebook most of the time it is not that important (for me).
